cartographer_ros/cartographer_ros/node.cc

```diff
@@ -194,7 +199,11 @@ void Node::PublishTrajectoryStates() {
     } else {
       // If we do not publish a new point cloud, we still allow time of the
       // published poses to advance.
-      stamped_transform.header.stamp = rclcpp::Time::now();
+      rclcpp::TimeSource ts(node_handle_);
```

Comment:
The time source should be persisted and possibly shared between all the clocks. I'd suggest possibly putting the time source as a member and initializing it with the node.